### PR TITLE
Stop if COSMO is used with periodic input

### LINKED ISSUE #827: signal SIGABRT error when running COSMO calculation
**Describe the bug**

The calculation with solvent use COSMO model was failed with error of
```
***  Geometry step: 0, Lattice step: 0

dftb+: malloc.c:3851: _int_malloc: Assertion `chunk_main_arena (bck->bk)' failed.

Program received signal SIGABRT: Process abort signal.

Backtrace for this error:
```
Then I have tried to run this calculation without solvent, it ran successfully.

**To Reproduce**

I run this calculation with commend of "dftb+ | tee output"

1. minimal example of how to trigger the bug (include self-contained input file)
[dftb_in.gen.txt](https://github.com/dftbplus/dftbplus/files/6901183/dftb_in.gen.txt)
[dftb_in.hsd.txt](https://github.com/dftbplus/dftbplus/files/6901184/dftb_in.hsd.txt)

2. I used the executables (x86_64/Linux), 21.1 to run this calculation. 
3. output showing the error
The error doesn't show in output file, the calculation just stopped.Here is the output file
[output.txt](https://github.com/dftbplus/dftbplus/files/6900990/output.txt)

**Expected behaviour**

The calculation was expected to run successfully or tell me what setting is wrong. I have no idea now about the error and how to solve it.

**Additional context**

The following is some information about Computer performance and system you may need:

Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                4
On-line CPU(s) list:   0-3
Thread(s) per core:    1
Core(s) per socket:    4
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 60
Model name:            Intel(R) Core(TM) i5-4590 CPU @ 3.30GHz
Stepping:              3
CPU MHz:               2448.815
CPU max MHz:           3700.0000
CPU min MHz:           800.0000
BogoMIPS:              6584.85
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              6144K
NUMA node0 CPU(s):     0-3
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm epb invpcid_single ssbd ibrs ibpb stibp kaiser tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid xsaveopt dtherm ida arat pln pts md_clear flush_l1d

Kernel: 4.9.0-15-amd64 #1 SMP Debian 4.9.258-1 (2021-03-08) x86_64 GNU/Linux
memory usage: 15958M

### PR DESCRIPTION
**Pull request by awvwgk**
Closes #827

### CODE REVIEW

File: src/dftbp/solvation/solvparser.F90 [hunk 1]
```diff
@@ -266,6 +266,10 @@ subroutine readSolvCosmo(node, geo, input)
     real(dp) :: temperature, shift, radScale
     type(TSolventData) :: solvent
 
+    if (geo%tPeriodic) then
+      call detailedError(node, "COSMO solvation currently not available with PBCs")
```
Comment (bhourahine) [inline src/dftbp/solvation/solvparser.F90:270]:
```suggestion
      call detailedError(node, "COSMO solvation currently not available with this geometrical boundary condition")
```

**Review by bhourahine — APPROVED** [review]
Looks reasonable, but should trap another BC.

File: src/dftbp/solvation/solvparser.F90 [hunk 2]
```diff
@@ -266,6 +266,10 @@ subroutine readSolvCosmo(node, geo, input)
     real(dp) :: temperature, shift, radScale
     type(TSolventData) :: solvent
 
+    if (geo%tPeriodic) then
```
[the remark this answers is not in the document]
Comment (awvwgk) [reply src/dftbp/solvation/solvparser.F90:269]: Wouldn't helical also set the `tPeriodic`?

[the remark this answers is not in the document]
Comment (bhourahine) [reply src/dftbp/solvation/solvparser.F90]: Not at the moment (see initGeometry_() ). This should get rationalised, so something like (.not.geo%molecular) or similar could be used.

**Pull request merged**